### ISSUE TITLE
[TODO] fix: BCELoss

### ISSUE THREAD

**Issue by L-M-Sherlock** [open-spaced-repetition/fsrs-optimizer-burn#17]
The current loss function is CrossEntropyLoss, which has already included a softmax layer inside. Its input is expected to contain the unnormalized logits for each class.

However, FSRS model's output is the stability, which is used to calculate the retention at certain `delta_t`. The retention is a form of probability. The softmax layer inside the CrossEntropyLoss will process the retention in an unexpected way. So the weights couldn't be optimized as expected.

Reference:
- https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html
- https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html
- https://medium.com/dejunhuang/learning-day-57-practical-5-loss-function-crossentropyloss-vs-bceloss-in-pytorch-softmax-vs-bd866c8a0d23
